source/agent/conference/conference.js

```diff
@@ -1891,58 +1928,70 @@ var Conference = function (rpcClient, selfRpcId) {
+              videoTrackIds.forEach(id => {
+                roomController.setPrimary(id, view);
+              });
+          } else {
+              roomController.setPrimary(activeInputStream, view);
           }
-        });
 
-        const mixedId = roomController.getMixedStream(view);
-        if (streams[mixedId] instanceof MixedStream) {
-          if (streams[mixedId].info.activeInput !== input) {
-            streams[mixedId].info.activeInput = input;
-            room_config.notifying.streamChange &&
-                sendMsg('room', 'all', 'stream',
-                    {id: mixedId, status: 'update', data: {field: 'activeInput', value: input}});
-          }
         }
-        callback('callback', 'ok');
-      } else {
-        const input = streams[activeInputStream] ?
-            activeInputStream : trackOwners[activeInputStream];
-        const activeAudioId = target.id;
-        if (streams[activeAudioId] instanceof SelectedStream) {
-          if (streams[activeAudioId].info.activeInput !== input ||
-              streams[activeAudioId].info.volume !== target.volume) {
-            streams[activeAudioId].info.activeInput = input;
-            streams[activeAudioId].info.activeOwner = target.owner;
-            streams[activeAudioId].info.volume = target.volume;
-            //TODO: separate major and activeInput when notifying
-            room_config.notifying.streamChange && sendMsg('room', 'all', 'stream',
+      });
+
+      const mixedId = roomController.getMixedStream(view);
+      if (streams[mixedId] instanceof MixedStream) {
+        if (streams[mixedId].info.activeInput !== input) {
+          streams[mixedId].info.activeInput = input;
+          room_config.notifying.streamChange &&
+              sendMsg('room', 'all', 'stream',
+                  {id: mixedId, status: 'update', data: {field: 'activeInput', value: input}});
+        }
+      }
+    } else {
+      const input = streams[activeInputStream] ?
+          activeInputStream : trackOwners[activeInputStream];
+      const activeAudioId = target.id;
+      if (streams[activeAudioId] instanceof SelectedStream) {
+        if (streams[activeAudioId].info.activeInput !== input ||
+            streams[activeAudioId].info.volume !== target.volume) {
+          streams[activeAudioId].info.activeInput = input;
+          streams[activeAudioId].info.activeOwner = target.owner;
+          streams[activeAudioId].info.volume = target.volume;
+          //TODO: separate major and activeInput when notifying
+          if (room_config.notifying.streamChange) {
+            sendMsg('room', 'all', 'stream',
+            {
+              id: activeAudioId,
+              status: 'update',
+              data: {field: 'activeInput', value: {id: input, volume: target.volume}}
+            });
+
+            !streams[activeAudioId].cascading && sendMsgTo('cascading', 'stream',
```

Comment:
If media processing such as mixing and transcoding always happen in local cluster, you don't need send these events.